src/wasm-s-parser.h

```diff
@@ -205,7 +205,7 @@ class SExpressionParser {
       input++;
       return allocator.alloc<Element>()->setString(IString(str.c_str(), false), dollared);
     }
-    while (input[0] && !isspace(input[0]) && input[0] != ')') input++;
+    while (input[0] && !isspace(input[0]) && input[0] != ')' && input[0] != '(') input++;
```

Comment:
If I remember correctly, `(module(memory ...)` doesn't parse unless you stop parsing `memory` when you see the `(`.